cpmpy/expressions/core.py

```diff
@@ -554,6 +554,72 @@ def value(self):
 
         return None # default
 
+    def get_bounds(self):
+        if self.is_bool():
+            return 0, 1 #boolean
+        elif self.name == 'mul':
+            lb1,ub1 = get_bounds(self.args[0])
+            lb2,ub2 = get_bounds(self.args[1])
+            bounds = [lb1 * lb2, lb1 * ub2, ub1 * lb2, ub1 * ub2]
+            return min(bounds), max(bounds)
+        elif self.name == 'sum':
+            lbs, ubs = zip(*[get_bounds(x) for x in self.args])
+            return sum(lbs), sum(ubs)
+        elif self.name == 'wsum':
+            weights, vars = self.args
+            var_bounds = np.array([get_bounds(arg) for arg in vars]).T
+            bounds = var_bounds * weights
+            return bounds.min(axis=0).sum(), bounds.max(axis=0).sum()  # for every column is axis=0...
+
+        elif self.name == 'sub':
+            lb1, ub1 = get_bounds(self.args[0])
+            lb2, ub2 = get_bounds(self.args[1])
+            bounds = [lb1 - lb2, lb1 - ub2, ub1 - lb2, ub1 - ub2]
+            return min(bounds), max(bounds)
+        elif self.name == 'div':
+            lb1, ub1 = get_bounds(self.args[0])
+            lb2, ub2 = get_bounds(self.args[1])
+            if lb2 <= 0 <= ub2:
+                raise ZeroDivisionError("division by domain containing 0 is not supported")
+            bounds = [lb1 // lb2, lb1 // ub2, ub1 // lb2, ub1 // ub2]
+            return min(bounds), max(bounds)
+        elif self.name == 'mod':
+            lb1, ub1 = get_bounds(self.args[0])
+            lb2, ub2 = get_bounds(self.args[1])
+            if lb2 <= 0 <= ub2:
+                raise ZeroDivisionError("% by domain containing 0 is not supported")
+            if ub1 <= 0 and ub2 < 0:
+                return lb1, 0
+            elif ub1 <= 0 and lb2 > 0:
+                return 0, ub2
+            elif lb1 >=0 and lb2 > 0:
+                return 0, ub1
```

Comment:
The above three cases need to be scrapped? It does not matter what `ub1`or `lb1` is, it only matters what `ub2` or `lb2` is. E.g., `-13%-2` is `-1`, so line 591 giving the bound `-13,0` seems overly permissive.